pkg/storage/replica.go

```diff
@@ -4286,15 +4303,28 @@ func shouldReplicaQuiesce(
 		}
 		return nil, false
 	}
+
 	var foundSelf bool
-	for id, progress := range status.Progress {
-		if id == status.ID {
+	for _, rep := range q.descRLocked().Replicas {
+		if uint64(rep.ReplicaID) == status.ID {
 			foundSelf = true
 		}
-		if progress.Match != status.Applied {
+		if progress, ok := status.Progress[uint64(rep.ReplicaID)]; !ok {
+			if log.V(4) {
+				log.Infof(ctx, "not quiescing: could not locate replica %d in progress: %+v",
+					rep.ReplicaID, progress)
+			}
+			return nil, false
+		} else if progress.Match != status.Applied {
+			// Skip any node in the descriptor which is not live.
+			if livenessMap != nil && !livenessMap[rep.NodeID] {
+				log.Infof(ctx, "skipping node %d because not live. Progress=%+v",
```

Comment:
Maybe I'm missing something, but wouldn't you also want to not send a message to that follower?

Reply:
I don't understand what you're suggesting. What message are you referring to that we shouldn't send to the follower, and what does this bit of code have to do with the message?

But thanks for pointing to this snippet of code because I realize the `log.Infof` need to be inside an `if log.V(4)` block.